sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -604,26 +614,55 @@ case class SortMergeJoinExec(
    * Generates the code for Inner join.
    */
   private def codegenInner(
+      ctx: CodegenContext,
       findNextJoinRows: String,
       beforeLoop: String,
       matchIterator: String,
       bufferedRow: String,
       conditionCheck: String,
       outputRow: String,
       eagerCleanup: String,
-      cleanedFlag: String): String = {
+      cleanedFlag: String,
+      matches: String): String = {
+    val processCurrentJoinRows = ctx.freshName("processCurrentJoinRows")
+    val processedCnt = ctx.freshName("processedCnt")
+    val checkBatchCode = if (batchSize > 0) {
+      s"""
+         |$processedCnt = $processedCnt + 1;
+         |if (($processedCnt & $batchSize) == 0) {
```

Comment:
Maybe we should use `$processedCnt > $batchSize` rather than `$processedCnt & $batchSize`, as if `batchSize = 8` and `the processedCnt = 1`, `$processedCnt & $batchSize` would get 0.

Reply:
+1

Reply:
I use another way to simplify the code.